src/qminer/qminer_core.cpp

```diff
@@ -5877,31 +5910,66 @@ void TIndex::IndexTextPos(const int& KeyId, const TStr& TextStr, const uint64& R
     IndexTextPos(KeyId, WordIdV, RecId);
 }
 
-void TIndex::IndexTextPos(const int& KeyId, const TUInt64V& WordIdV, const uint64& RecId) {
+void TIndex::UpdateTextPos(const int& KeyId, const TUInt64V& WordIdV, const uint64& RecId, void (TGix<TQmGixKey, TQmGixItemPos>::*UpdateMethod)(const TQmGixKey&, const TQmGixItemPos&)) {
```

Comment:
Replace function pointer with vector `TVec<TKeyDat<TKeyWord, TQmGixItemPos> >&`. Do actuall interaction with Gix in `IndexTextPos` or `DeleteTextPos`. Makes code much simpler and easier to comprehend :-)